packages/loaders/json-schema/src/getComposerFromJSONSchema.ts

```diff
@@ -851,6 +850,24 @@ export function getComposerFromJSONSchema(
             });
           }
         }
+
+        let allOf;
+        if (subSchema.allOf) {
+          allOf = subSchema.allOf.map((schema: any) => {
```

Comment:
I think this manipulates the elements of allOf. Let's say I have the following schema. This makes `Foo.id` and `Bar.id` required, and when I use `Foo` and `Bar`, they will have `id` required but actually they are not.
```json
{
  "definitions": {
    "Foo": {
      "type": "object",
      "properties": {
        "id": {
          "type": "string"
        }
      }
    },
    "Bar": {
      "type": "object",
      "properties": {
        "id": {
          "type": "string"
        }
      }
    },
    "Baz": {
      "allOf": [
        {
          "$ref": "#/definitions/Foo"
        },
        {
          "$ref": "#/definitions/Bar"
        }
      ],
      "required": [
        "id"
      ]
    }
  }
}
```

Reply:
No i actually check if $resolvedRef is set and then don't modify the schema. So it only updates local schema's.

Reply:
@ardatan tests are failing because some snapshots have changed and it actually fixes existing tests. I've posted more details in slack.

Reply:
But then it means this fix won't work if there is a referenced schema 🤔

Reply:
@ardatan no this now actually fixes issues in the cloudflare schemas. see:

![Screenshot 2023-11-03 at 11 45 21](https://github.com/ardatan/graphql-mesh/assets/32583/e73f3438-425e-42d9-88fb-b1950506ee8e)
![Screenshot 2023-11-03 at 11 42 20](https://github.com/ardatan/graphql-mesh/assets/32583/44af682e-9881-41f5-9e52-50b194bff77d)